torch/csrc/jit/passes/quantization.cpp

```diff
@@ -676,5 +676,246 @@ TORCH_API script::Module InsertObservers(
+    return qparams;
+}
+
+void QuantizeHelper::quantizeBias(Value* v) {
+  // Traverse to the place where this is used
+  Node* aten_call_node = nullptr;
+  std::vector<std::string> ops_with_bias = {"aten::conv2d", "aten::_convolution"};
+  for (const auto& use: v->uses()) {
+    if (std::find(ops_with_bias.begin(), ops_with_bias.end(),
+                  use.user->kind().toQualString()) != ops_with_bias.end()) {
+      // Make sure there is no observer module for bias
+      c10::optional<script::Module> observer_module;
+      auto observer_name = findObserverName(v);
+      TORCH_INTERNAL_ASSERT(!observer_name,
+                            "bias should not be observed!");
+      Value* activation = use.user->inputs()[0];
+      Value* weight = use.user->inputs()[1];
+      // Get qparam from activation
+      IValue act_qparam = getQParam(activation);
+      // Get qparam from weight
+      IValue weight_qparam = getQParam(weight);
+      IValue bias_scale =  at::scalar_tensor(
+          c10::Scalar(act_qparam.toTuple()->elements()[0].toTensor().item().toDouble() *
+                      weight_qparam.toTuple()->elements()[0].toTensor().item().toDouble()),
+          at::kDouble);
+      IValue bias_qparam = c10::ivalue::Tuple::create(
+          std::vector<IValue>({bias_scale,
+                               at::scalar_tensor(c10::Scalar(0))}), act_qparam.toTuple()->type);
+      Node* dequant = insertQuantDeQuantCall(v, bias_qparam, at::kQInt32);
+      v->replaceAllUsesWith(dequant->output());
+      Node* q = traverseToQuantNode(dequant);
+      TORCH_INTERNAL_ASSERT(q != nullptr);
+      q->replaceInputWith(dequant->output(), v);
+      break;
+    }
+  }
+}
+
+void QuantizeHelper::quantizeTensor(Value* v,
+                                    bool insert_after) {
+  auto observer_name = findObserverName(v);
+  if (!observer_name) {
+    return;
+  }
+  auto observer_module = module_.find_module(observer_name.value());
+  TORCH_INTERNAL_ASSERT(observer_module,
+                        "Expect observer module " +                     \
+                        observer_name.value() + " in quantizeTensor");
+  // remove observer_module
+  observer_modules_to_remove_.push_back(observer_name.value());
+  // remove observer forward call
+  for (const Use& u: v->uses()) {
+    Node* user = u.user;
+    if (user->kind() == prim::CallMethod &&
+        user->s(c10::attr::name) == "forward" &&
+        user->inputs()[0]->node()->kind() == prim::GetAttr &&
+        user->inputs()[0]->node()->s(attr::name) == observer_name) {
+      // Observer forward call node
+      nodes_to_destroy_.push_back(user);
+      // GetAttr node for observer module
+      nodes_to_destroy_.push_back(user->inputs()[0]->node());
+    }
+  }
+
+  // calculate qparams and insert quant/dequant calls
+  auto calculate_qparams = observer_module.value().get_method("calculate_qparams");
+  IValue qparams = calculate_qparams(std::vector<IValue>());
```

Comment:
Can we use `getQParam` function here? Seems very similar to what it is doing anyway.

Reply:
yeah, actually I can use a flag saying whether we want to remove the observer_module/node to merge the previous code to getQParam as well

Reply:
I'm not a big fan of adding flags to functions - if we do this, `getQParam` name would no longer reflect what it does, which is not great. It should be renamed to something like `getQParamAndRemoveObserverCall`, which I think would be better split into two separate functions. And this is basically what we currently have (except that the code for `removeObserverCall` isn't factored out to a separate function).

Reply:
ok, it's just we'll findObserverName multiple times in that case, but it should be fine still